arch/arm/boot/dts/armada-385-ts7840.dts

```diff
@@ -0,0 +1,840 @@
+		"MPP11_GE_TXCTL",
+		"MPP12_GE_RXD0",
+		"MPP13_GE_RXD1",
+		"MPP14_GE_RXD2",
+		"MPP15_GE_RXD3",
+		"MPP16_GE_RXCTL",
+		"MPP17_GE_RXCLK",
+		"WIFI_IRQ_N",
+		"FPGA_RESET",
+		"",
+		"MPP_21_GE_RXD0",
+		"SPI_0_MOSI",
+		"SPI_0_CLK",
+		"SPI_0_MISO",
+		"SPI_0_BOOT_CS0#",
+		"",
+		"MPP_27_GE_TXCLK",
+		"MPP_28_GE_TXD0",
+		"MPP_29_GE_TXD1",
+		"MPP_30_GE_TXD2",
+		"MPP_31_GE_TXD3";
+};
+
+&gpio1 {
+	gpio-line-names =
+		"MPP_32_GE_TXCTL",
+		"CPU_SPEED_0",
+		"CPU_SPEED_3",
+		"CPU_SPEED_4",
+		"CPU_TYPE_0",
+		"MPP_37_GE_RXCLK",
+		"MPP_38_GE_RXD1",
+		"MPP_39_GE_RXD2",
+		"MPP_40_GE_RXD3",
+		"MPP_41_GE_RXCTL",
+		"GPS_PPS",
+		"ETH_SW_INT#",
+		"CPU_TYPE_1",
+		"",
+		"",
+		"",
+		"",
+		"FPGA_FLASH_SELECT",
+		"EMMC_CMD",
+		"SPREAD_SPECTRUM#",
+		"",
+		"DETECT_9478",
+		"EMMC_D3",
+		"EMMC_D0",
+		"",
+		"EMMC_CLK",
+		"EMMC_D1",
+		"EMMC_D2",
+		"",
+		"",
+		"",
+		"";
+};
+
+&i2c0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c0_pins>;
+	clock-frequency = <100000>;
+	status = "okay";
+
+	silabs_wdt: silabs-wdt@54 {
+		compatible = "technologic,ts7100-wdt";
+		reg = <0x54>;
+	};
+
+	rtc1: m41t00s@68 {
+		compatible = "m41t00";
+		reg = <0x68>;
+	};
+};
+
+&mdio {
+	pinctrl-names = "default";
+	pinctrl-0 = <&mdio_pins>;
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+		/* Configure LED[0] to blink for activity */
+		marvell,reg-init = <3 16 0 0x1aa4>;
+	};
+
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+		/* Configure LED[0] to blink for activity */
+		marvell,reg-init = <3 16 0 0x1aa4>;
+	};
+
+	switch@2 {
+		compatible = "marvell,mv88e6085";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&switch_pins0>;
+
+		#address-cells = <1>;
+		#size-cells = <0>;
+		dsa,member = <0 0>;
+		reg = <2>;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			ports@0 {
+				reg = <0>;
+				label = "lan0";
+			};
+
+			ports@1 {
+				reg = <1>;
+				label = "lan1";
+			};
+
+			ports@2 {
+				reg = <2>;
+				label = "lan2";
+			};
+
+			ports@3 {
+				reg = <3>;
+				label = "lan3";
+			};
+
+			ports@4 {
+				reg = <4>;
+				label = "lan4";
+			};
+
+			port@5 {
+				reg = <5>;
+				label = "cpu";
+				ethernet = <&eth2>;
+				fixed-link {
+					speed = <1000>;
+					full-duplex;
+				};
+			};
+		};
+	};
+};
+
+&pciec {
+	status = "okay";
+};
+
+/* Mini PCIe */
+&pcie2 {
+	status = "okay";
+};
+
+/* FPGA */
+&pcie3 {
+	status = "okay";
+};
+
+&pinctrl {
+	sdhci_pins0: sdhci-pins0 {
+		marvell,pins = "mpp50", "mpp54", "mpp55",
+			       "mpp57", "mpp58", "mpp59";
+		marvell,function = "sd0";
+	};
+
+	switch_pins0: switch-pins0 {
+		marvell,pins = "mpp43";
+		marvell,function = "gpio";
+	};
+};
+
+/* We do not use the CPU's built in RTC */
+&rtc {
+	status = "disabled";
+};
+
+&sdhci {
+	bus-width = <4>;
+	no-1-8-v;
+	pinctrl-0 = <&sdhci_pins0>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&spi0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi0_pins>;
+	status = "okay";
+	num-cs = <3>;
+	cs-gpios = <0>, <&fpga_bank1 9 0>;
```

Comment:
Replace hardcoded `0` with `GPIO_ACTIVE_LOW` (LOW needed for SPI CS unless it really is active high, then it requires another property added)

Reply:
Fixed in latest